modules/biobank/php/specimendao.class.inc

```diff
@@ -0,0 +1,719 @@
+                'typeId'            => intval($data['SpecimenTypeID']),
+                'quantity'          => floatval($data['Quantity']),
+                'unitId'            => intval($data['UnitID']),
+                'fTCycle'           => intval($data['FreezeThawCycle']),
+                'parentSpecimenIds' => $parentSpecimenIds,
+                'candidateId'       => intval($data['CandidateID']),
+                'sessionId'         => intval($data['SessionID']),
+                'poolId'            => intval($data['PoolID']),
+                'collection'        => array(
+                                        'quantity' => floatval($data['CollectionQuantity']),
+                                        'unitId'   => intval($data['CollectionUnitID']),
+                                        'centerId' => intval($data['CollectionCenterID']),
+                                        'date'     => $data['CollectionDate'],
+                                        'time'     => $data['CollectionTime'],
+                                        'comments' => $data['CollectionComments'],
+                                        'data'     => json_decode($data['CollectionData'], true),
+                                       ),
+                'preparation'       => array(
+                                        'protocolId' => intval($data['SpecimenProtocolID']),
+                                        'centerId'   => intval($data['PreparationCenterID']),
+                                        'date'       => $data['PreparationDate'],
+                                        'time'       => $data['PreparationTime'],
+                                        'comments'   => $data['PreparationComments'],
+                                        'data'       => $data['PreparationData'] ? json_decode($data['PreparationData'], true) : null,
+                                       ),
+                'analysis'          => array(
+                                        'methodId' => intval($data['SpecimenMethodID']),
+                                        'centerId' => intval($data['AnalysisCenterID']),
+                                        'date'     => $data['AnalysisDate'],
+                                        'time'     => $data['AnalysisTime'],
+                                        'comments' => $data['AnalysisComments'],
+                                        'data'     => $data['AnalysisData'] ? json_decode($data['AnalysisData'], true) : null,
+                                       ),
+               );
+    }
+
+    /**
+     * Instatiates a Specimen Object using the values sent from the front-end.
+     *
+     * @param array $data Data to be used to instantiate the Specimen Object.
+     *
+     * @return Specimen $specimen Instantiated Specimen Object
+     */
+    public function getInstanceFromArray(array $data) : Specimen
+    {
+        $specimen = $this->getInstance();
```

Comment:
This function looks like a constructor. Maybe it would be better to add this to Specimen if it makes sense

```php
public function __construct(array $data) {
    foreach($data as $name => $value) {
         $this->$name = $value;
    }
}
```

and then this function can just be

```
public function getInstanceFromArray(array $data) : Specimen
{
    return new Specimen($data);
}

Reply:
Definitely a possibility! We can discuss this in more detail at our meeting next Tuesday.

@driusan @ridz1208 @xlecours